sdk/storage/azure-storage-file-share/azure/storage/fileshare/_file_client.py

```diff
@@ -1115,31 +1159,22 @@ def get_ranges(  # type: ignore
         :paramtype lease: ~azure.storage.fileshare.ShareLeaseClient or str
         :keyword int timeout:
             The timeout parameter is expressed in seconds.
-        :returns: A list of valid ranges.
-        :rtype: List[dict[str, int]]
+        :returns:
```

Comment:
The return type needs to be changed.
I'm thinking if we want to create a new API to get range diff, sharing the same API with get_ranges could cause confusion I feel.
Since `# type: (...) -> Union[List[Dict[str, int]], Tuple[List[Dict[str, int]], List[Dict[str, int]]]]` is kinda confusing. Users might not sure if they will get a Tuple or List.
@annatisch @kasobol-msft Do you have any opinion?

Reply:
Thanks @xiafu-msft - yes we shouldn't have mixed return types.
So are "cleared ranges" only available when the diff snapshot parameter is provided?

Reply:
Yes, that is correct